WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationMemberJoinEventProcessor.swift

```diff
@@ -32,9 +35,104 @@ protocol ConversationMemberJoinEventProcessorProtocol {
 
 struct ConversationMemberJoinEventProcessor: ConversationMemberJoinEventProcessorProtocol {
 
-    func processEvent(_: ConversationMemberJoinEvent) async throws {
-        // TODO: [WPB-10168]
-        assertionFailure("not implemented yet")
+    let context: NSManagedObjectContext
+    let conversationRepository: any ConversationRepositoryProtocol
+    let userRepository: any UserRepositoryProtocol
+
+    func processEvent(_ event: ConversationMemberJoinEvent) async throws {
+        let conversationID = event.conversationID
+        let id = conversationID.uuid
+        let domain = conversationID.domain
+
+        var conversation = await conversationRepository.fetchConversation(
+            with: id,
+            domain: domain
+        )
+
+        if conversation == nil {
+            // Sync conversation
+            try await conversationRepository.pullConversation(with: conversationID)
+            conversation = await conversationRepository.fetchConversation(
+                with: id,
+                domain: domain
+            )
+        }
+
+        guard let conversation else {
+            return WireLogger.eventProcessing.error(
```

Comment:
add eventId and conversationId as attributes

Reply:
done in https://github.com/wireapp/wire-ios/pull/2041/commits/b6b6339287cb7d599f030abff366b70272167cc9, added the conversationID, we no longer have an event ID in the refactored code.

Reply:
ok, I guess we'll have to revisit logging anyway in order to track vitals like number of processed events type X or Y